### ISSUE TITLE
Library Application Type

### ISSUE THREAD

**Issue by musketyr** [micronaut-projects/micronaut-starter#700]
it would be great if there is a "Micronaut Library" option in the "Application Type"  to generate a Micronaut library

**Comment by sdelamo** [comment]
This will involve: 

- Creating a new entry in `ApplicationType`
- Define how the build for a library would look like in both Maven and Gradle and Java, Google, and Kotlin. We could initially support only Gradle and Java for Library authors to reduce scope. 

Generated application characteristics: 

- The build should contain the necessary configuration to publish to Maven Central. I am using [such a configuration](https://github.com/sdelamo/eurorates/blob/master/build.gradle#L59-L137). 
- The build should start with the minimum dependencies for library authors. For my pet projects, I have used [`inject-java`, `validation` and `runtime`](https://github.com/sdelamo/eurorates/blob/master/build.gradle#L23-L30) and the [test framework](https://github.com/sdelamo/eurorates/blob/master/build.gradle#L41-L43). I have not used Micronaut Test, but we should use it for the `Library` application type. 
- We should create a guide showing how to use the profile and demonstrate good practices. E.g., don't expose interfaces using reactive library-specific types (`Mono`, `Single`, ...) but prefer [`Publisher` with `SingleResult`](https://github.com/sdelamo/eurorates/blob/master/src/main/java/com/softamo/eurorates/EuroRatesClient.java#L46-L47) instead. 
- Generated application should contain instructions in the README file with links to  Sonatype. For example: [Publish Guide](https://central.sonatype.org/publish/publish-guide/)
- In the `README.md` file we should prompt people to submit PRs to include their library in starter with the [community flag](https://github.com/micronaut-projects/micronaut-starter/blob/3.3.x/starter-core%2Fsrc%2Fmain%2Fjava%2Fio%2Fmicronaut%2Fstarter%2Ffeature%2Fcamunda%2FCamunda.java#L54-L57).

**Comment by sdelamo** [comment]
I think we should preconfigure the builds of the generated applications to use https://jreleaser.org

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?

**Comment by musketyr** [comment]
yes, the latest Micronaut Launch website still does not offer the Library type
